### PR TITLE
Enhancement/223 saf sdk to saf docs

### PR DESCRIPTION
**Pull request by msalak-scia**
new part of documentation which should describe the usage of software development kit
it contains 4 new pages

### CODE REVIEW

**Comment by dschuermans** [comment]
I might have phrased things wrong originally.
I did say you could use the markdown files from the nuget packages but looking at this now, it doesn't quite fit.

We'd need the following:
- [ ] Introduction page (what is the SDK, what is it used for)
- [ ] Available packages and their dependencies
- [X] Per package a page detailing what it does (which we have now)

File: docs/SDK/StructuralAnalysisFormat.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+# StructuralAnalysisFormat
+This package implements the [Structural Analysis Format](https://www.saf.guide) in C#
```
Comment (dschuermans) [inline docs/SDK/StructuralAnalysisFormat.md:2]: The link here can be removed

File: docs/SDK/StructuralAnalysisFormat.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+## Features
+- Straight forward API (One import and one export service)
+- Validation
+- Compatibility support between versions
+- Effortlessly switch between metric & imperial values
+
+## Dependencies
+- Uses EPPlus (4.5.3.3) for handling excel files
+- Uses FluentValidation for validation
+- Uses UnitsNet for unit operations
```
Comment (dschuermans) [inline docs/SDK/StructuralAnalysisFormat.md:15]: For each package, add a link to their respective nuget.org page

File: docs/SDK/StructuralAnalysisFormat.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+## What do you need to do?
+In order to use this SDK, it is sufficient for you to provide a mapping between the data objects in your software and the data objects defined in the SDK
+Once you have that, you only need to call the import or export service and the rest is taken care of by the SDK.
+
+## Remarks
+Due to how this SDK is defined, with most of the actual implementation details being internal, you should also use one of our bootstrapping packages:
+- **StructuralAnalysisFormat.Bootstrappers.SimpleInjector4**  
+  _If you wish to use SimpleInjector v4_
+- **StructuralAnalysisFormat.Bootstrappers.SimpleInjector5**  
+  _If you wish to use SimpleInjector v5_
```
Comment (dschuermans) [inline docs/SDK/StructuralAnalysisFormat.md:26]: Add links here to the corresponding pages

File: docs/SDK/StructuralAnalysisFormat.Tests.Infrastructure.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+## Data builders
+One of the features are what we call **data builders**
+These are fluent-api classes that allow you to quickly construct data objects that by default pass validation.
+They also have the benefit that if a property would be renamed on an object, the data builder will be adjusted but the API of the builder remains the same.
+Which means you won't have to make any changes to your test code!
+
+### Good to know - Metric & Imperial
+Any values unit wise that are set using doubles, will always be metric as this is how the data builders are set up.
+If the builder accepts actual units, you are free to provide it with units defined in the imperial system for example.
```
Comment (dschuermans) [inline docs/SDK/StructuralAnalysisFormat.Tests.Infrastructure.md:12]: Turn this into one of those tip boxes

File: docs/SDK/StructuralAnalysisFormat.Tests.Infrastructure.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+### Good to know - Sane values
+While we do our best to make sure that the model actually makes sense in the context of structural engineering, this may sometimes not be the case.  
+So it may happen that some objects constructed by either the data builders or from the house model have weird values that do not make sense to a structural engineer.
```
Comment (dschuermans) [inline docs/SDK/StructuralAnalysisFormat.Tests.Infrastructure.md:16]: Turn this into one of those tip boxes

File: docs/SDK/StructuralAnalysisFormat.Bootstrappers.SimpleInjector4.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
+# StructuralAnalysisFormat.Bootstrappers.SimpleInjector4
+This package will bootstrap the internal components for you, using SimpleInjector v4
```
Comment (dschuermans) [inline docs/SDK/StructuralAnalysisFormat.Bootstrappers.SimpleInjector4.md:2]: Add a link to the SimpleInjector 4 docs

File: docs/SDK/StructuralAnalysisFormat.Bootstrappers.SimpleInjector5.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
+# StructuralAnalysisFormat.Bootstrappers.SimpleInjector4
+This package will bootstrap the internal components for you, using SimpleInjector v5
```
Comment (dschuermans) [inline docs/SDK/StructuralAnalysisFormat.Bootstrappers.SimpleInjector5.md:2]: Add a link to the SimpleInjector 5 docs

**Review by dschuermans — CHANGES_REQUESTED** [review]
Please make the requested changes 😄

File: docs/SDK/StructuralAnalysisFormat.md (hunk 3 above)
Comment (dschuermans) [reply docs/SDK/StructuralAnalysisFormat.md:26]: v4: https://docs.simpleinjector.org/en/4.0/
v5: https://docs.simpleinjector.org/en/5.0/

File: docs/SDK/StructuralAnalysisFormat.Bootstrappers.SimpleInjector4.md (hunk 6 above)
Comment (dschuermans) [reply docs/SDK/StructuralAnalysisFormat.Bootstrappers.SimpleInjector4.md:2]: https://docs.simpleinjector.org/en/4.0/

File: docs/SDK/StructuralAnalysisFormat.Bootstrappers.SimpleInjector5.md (hunk 7 above)
Comment (dschuermans) [reply docs/SDK/StructuralAnalysisFormat.Bootstrappers.SimpleInjector5.md:2]: https://docs.simpleinjector.org/en/5.0/

**Comment by msalak-scia** [comment]
Thank you @dschuermans for the detailed requests. They should be hopefully all solved. I also added the link to nuget.org for every package, is it ok or does it just make it more complicated :D?

**Review by dschuermans — APPROVED** [review]

**Pull request merged**